### PR TITLE
Simple pattern support + integer ranges

### PR DESCRIPTION
**Pull request by joennlae**
This PR introduces simple regex pattern support + integer ranges.

### CODE REVIEW

**Comment by Ubospica** [comment]
@joennlae Thanks for your contribution! Sorry for not replying to you in time, and I will review the code later today

File: cpp/json_schema_converter.cc [hunk 1]
```diff
@@ -661,12 +779,38 @@ std::string JSONSchemaConverter::VisitInteger(
       schema,
       {
           "multipleOf",
-          "minimum",
```
Comment (Ubospica) [inline cpp/json_schema_converter.cc:664]: Could you add tests for minimum and maximum as well?

File: cpp/json_schema_converter.cc [hunk 2]
```diff
@@ -652,6 +654,122 @@ std::string JSONSchemaConverter::VisitAny(
          kBasicArray + " | " + kBasicObject;
 }
 
+std::string generateRangeRegex(std::optional<int> start, std::optional<int> end) {
```
Comment (Ubospica) [inline cpp/json_schema_converter.cc:657]: Could you make generateRangeRegex a static member function of JSONSchemaConverter, and follow the naming conversion (GenerateRangeRegex)?

File: cpp/json_schema_converter.cc [hunk 3]
```diff
@@ -661,12 +779,38 @@ std::string JSONSchemaConverter::VisitInteger(
...
+        double end_double = schema.at("exclusiveMaximum").get<double>();
+        end = static_cast<int>(end_double);
+      }
+      range_regex = generateRangeRegex(start, end);
+    }
+    if (!range_regex.empty()) {
+      std::string converted_regex = RegexToEBNF(range_regex, false);
+      return converted_regex;  // not " " for numbers
+    }
+  } catch (const std::exception& e) {
```
Comment (Ubospica) [inline cpp/json_schema_converter.cc:811]: Please remove the usage of try-catch block and std::exception. The reason is we maintain a typescript binding running in WASM, which has not supported exceptions yet. Just throw an error at the error line.

File: cpp/json_schema_converter.cc [hunk 4]
```diff
@@ -698,10 +842,19 @@ std::string JSONSchemaConverter::VisitString(
       {
           "minLength",
           "maxLength",
-          "pattern",
           "format",
```
Comment (Ubospica) [inline cpp/json_schema_converter.cc:845]: Could you add support for `format` keyword as well? We can support `ipv4`, `ipv6`, `date-time`, `date`, `time`, `email` now. Check out https://github.com/mlc-ai/xgrammar/blob/main/tests/python/test_regex_converter.py, and https://json-schema.org/understanding-json-schema/reference/string.

**Review by Ubospica — CHANGES_REQUESTED** [review]
Overall look great to me. Thanks for your contribution!

**Comment by Ubospica** [comment]
We can merge this PR now, and I will update according to the above comments in a new PR. Thanks @joennlae!
